consulate/cli.py

```diff
@@ -267,19 +282,76 @@ def register(consul, args):
     'rm': kv_rm,
     'set': kv_set}
 
+def run_once(consul, args):
+    """Ensure only one process can run a command at a time
+
+    :param consulate.api_old.Consul consul: The Consul instance
+    :param argparser.namespace args: The cli args
+
+    """
+    try:
+        import time
+        import subprocess
+
+        session = consul.session
+        if not consul.kv.acquire_lock(args.prefix, session):
+            sys.stdout.write('Cannot obtain the required lock. Exiting\n')
+            sys.exit(2)
+        if args.interval:
+            now = int(time.time())
+            last_run = consul.kv.get(args.prefix)
+            if str(last_run) not in ['null', 'None'] and \
+                int(last_run) + int(args.interval) > now:
+                sys.stdout.write("Last run happened fewer than "\
+                    "{0} second ago. Exiting\n".format(args.interval))
+                return
+            consul.kv[args.prefix] = now
+
+        # Should the subprocess return an error code, release the lock
+        try:
+            error = False
+            subprocess.check_output(args.operation, stderr=subprocess.STDOUT)
+        # If the subprocess fails
+        except subprocess.CalledProcessError as e:
+            sys.stdout.write('"{0}" exited with return code "{1}" and output {2}\n'.format(
+                args.operation, e.returncode, e.output))
+            error = True
+        # If the command doesn't exist
+        except OSError as e:
+            sys.stdout.write('"{0}" command does not exist\n'.format(args.operation))
+            error = True
+        # Otherwise
+        except Exception as e:
+            sys.stdout.write('"{0}" exited with error "{1}"\n'.format(
+                args.operation, e))
+            error = True
+
+        consul.kv.release_lock(args.prefix, session)
+        if error:
+            sys.exit(1)
+    except exceptions.ConnectionError:
+        connection_error()
 
 def main():
     """Entrypoint for the consulate cli application"""
     args = parse_cli_args()
 
     if args.api_scheme == 'http+unix':
-        adapter = None
-    else:
         adapter = adapters.UnixSocketRequest
+        port = None
+        api_host = os.environ.get('CONSUL_HTTP_ADDR').replace('unix://', '')
```

Comment:
I'm trying to simplify the command line arguments when connecting with a unix socket. If we're using sockets, the idea is to read the socket location from Consul's stored environment variable, instead of requiring it over the command line.

I could use the environment variable as the default and override if it's passed in as an arg